index.js

```diff
@@ -52,6 +53,10 @@ module.exports = options => {
 			if (props.type) {
 				const type = props.type;
 
+				if (availableTypes.indexOf(type) === -1) {
+					throw new TypeError(`Expected "${key}" to be boolean or string, got ${type}`);
```

Comment:
Could you change this to `Expected "${key}" to be one of ["string", "boolean", "number", "object"], got ${type}`?